ometiff-JavaScript/PixelSizeConsistency.js

```diff
@@ -58,24 +62,45 @@ getExif().then(function(result) {
             "PhysicalSizeYUnit" : physicalSizeYUnit,
             "PhysicalSizeZ" : physicalSizeZ,
             "PhysicalSizeZUnit" : physicalSizeZUnit,
-            "DimensionOrder" : output['OME']['Image'][0]['Pixels'][0]['$']['DimensionOrder']
+            "DimensionOrder" : output['OME']['Image'][0]['Pixels'][0]['$']['DimensionOrder'],
         }
         console.table(metadata)
         
         let jsonData = require('./test_image.json')
+
         let pixelSize = jsonData['PixelSize']
         let physicalSizeUnit = jsonData['PixelSizeUnits']
 
         let factorX = convertFactor(physicalSizeXUnit, physicalSizeUnit);
         let factorY = convertFactor(physicalSizeYUnit, physicalSizeUnit);
         let factorZ = convertFactor(physicalSizeZUnit, physicalSizeUnit);
 
+
+        // physicalSizeUnit consistency check
         if (physicalSizeX * factorX !== pixelSize[0] || physicalSizeY * factorY !== pixelSize[1] || physicalSizeZ * factorZ !== pixelSize[2]) {
             console.log("\x1b[31m%s\x1b[0m", "PixelSize is not consistent with PhysicalSizeX, PhysicalSizeY and PhysicalSizeZ OME metadata fields")
             console.log("\x1b[31mpixelSize in JSON is %s, the converted physicalSize from OME-XML is [%s, %s, %s]\x1b[0m", pixelSize, physicalSizeX * factorX, physicalSizeY * factorY, physicalSizeZ * factorZ)
         }else{
             console.log("\x1b[36m%s\x1b[0m", "PixelSize is consistent with PhysicalSizeX, PhysicalSizeY and PhysicalSizeZ OME metadata fields")
         }
 
+
+        // optional fields consistency check
+        let fields = {'Immersion': 'Immersion', 'NumericalAperture': 'LensNA', 'Magnification': 'NominalMagnification'}
+
+        let objective = output['OME']['Instrument'][0]['Objective'][0]['$']
```

Comment:
The "Instrument" of "Objective" field may not be present at all in the OME-TIFF so I think this just need an "if present do something else None" kind of behavior (don't know how that works in JS).

Reply:
Done